### PR TITLE
fix: Line list and pivot table (enrollment) do not give the same total [2.37-DHIS2-11524-patch.2] - bug fix (regression)

### PR DESCRIPTION
**Pull request by d-bernat**

### CODE REVIEW

**Review by maikelarabori — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventDataQueryService.java [hunk 1]
```diff
@@ -224,7 +224,16 @@ public EventQueryParams getFromRequest( EventDataQueryRequest request )
 
         if ( request.getOutputType() == EventOutputType.ENROLLMENT )
         {
-            params.withStartEndDatesForPeriods( false );
+            if ( request.getStartDate() == null || request.getEndDate() == null )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventDataQueryService.java:227]: You can also use `!request.hasStartEndDate()` for more a expressive syntax.

**Comment by larshelge** [comment]
Would be good with a more descriptive PR name/description.

**Comment by d-bernat** [comment]
> Would be good with a more descriptive PR name/description.

Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventDataQueryService.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventDataQueryService.java:227]: Done

**Review by larshelge — APPROVED** [review]

**Pull request merged**